pkgs/native_assets_builder/test_data/native_dynamic_linking/hook/build.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:logging/logging.dart';
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+void main(List<String> args) async {
+  await build(args, (config, output) async {
+    final logger = Logger('')
+      ..level = Level.ALL
+      ..onRecord.listen((record) => print(record.message));
+
+    final builders = [
+      CBuilder.library(
+        name: 'debug',
+        assetName: 'debug',
+        sources: [
+          'src/debug.c',
+        ],
+      ),
+      CBuilder.library(
+        name: 'math',
+        assetName: 'math',
+        sources: [
+          'src/math.c',
+        ],
+        // TODO(https://github.com/dart-lang/native/issues/190): Use specific
+        // API for linking once available.
+        flags: config.dynamicLinkingFlags('debug'),
+      ),
+      CBuilder.library(
+        name: 'add',
+        assetName: 'add.dart',
+        sources: [
+          'src/add.c',
+        ],
+        // TODO(https://github.com/dart-lang/native/issues/190): Use specific
+        // API for linking once available.
+        flags: config.dynamicLinkingFlags('math'),
+      )
+    ];
+
+    // Note: This builders need to be run sequentially because they depend on
+    // each others output.
+    for (final builder in builders) {
+      await builder.run(
+        config: config,
+        output: output,
+        logger: logger,
+      );
+    }
+  });
+}
+
+extension on BuildConfig {
```

Comment:
Code like this, switching over OSs, seems like something which should actually live in a helper package. Or is it specific to this example?

Reply:
Or is this covered by https://github.com/dart-lang/native/issues/190?